### PR TITLE
Feature/envoy & xds server graceful exit

### PR DESCRIPTION
**Pull request by wanlin31**
This PR is to gracefully shutdown the sidecar and the xds server when the main test process terminate. The PR is based on #235.

### CODE REVIEW

File: containers/runtime/envoy/graceful_exit.sh
Comment (gnossen) [inline containers/runtime/envoy/graceful_exit.sh:13]: Nit: Shell variables are traditionally `ALL_CAPS`.

**Review by gnossen — APPROVED** [review]
Looks good. LGTM to the contents of this PR but not necessarily the diff base.

File: containers/runtime/xds/cmd/main.go
Comment (paulosjca) [inline containers/runtime/xds/cmd/main.go:1]: Copyright. Check other files as well.

File: containers/runtime/xds/logger.go
Comment (paulosjca) [inline containers/runtime/xds/logger.go:11]: "prints", "stops", in all functions.

**Comment by wanlin31** [comment]
@gnossen Hi Richard, given this is already a complicate pr, I will add the validation in a follow up PR once merged. This PR is pretty much the same as #235 with changes we discussed in the meeting. Thank you!

**Review by gnossen — APPROVED** [review]

File: containers/runtime/xds/config/suite_test.go
Comment (wanlin31) [reply containers/runtime/xds/config/suite_test.go:2]: Thank you for helping me get this right!

File: proto/testupdater/test_updater.proto
Comment (wanlin31) [reply proto/testupdater/test_updater.proto:19]: Got it, I reverted the package name back. I used to think the package name should reflect the fact that we also communicate if the test is a proxied test. I will also create a separate PR for doc.go. Thank you.

**Review by paulosjca — APPROVED** [review]
LGTM, but please wait for release 1.0.0 to be complete before merging.

**Pull request merged**
